tools/knb/README.md

```diff
@@ -0,0 +1,112 @@
+# knb - kn builder
+
+CLI tool to enhance plugin build experience for [Knative Client](https://github.com/knative/client).
+
+
+### Build
+```bash
+go build
+```
+
+### Install
+```bash
+go get knative.dev/client/tools/knb
+```
+
+### Usage
+
+##### Create custom `kn` distribution
+
+The `knb` can be used to generate enhanced `kn` source files with inlined plugins.
+
+Create configuration file `.kn.yaml` in a root directory of `knative/client` that should specify at least `name, module, version` coordinates of the plugin.
+The `plugin distro` will generated the required go files and add dependency to `go.mod`.
+
+Required:
+* name
+* module - go module name to be used for import and in go.mod file
+* version - accepted values are git tag or branch name of go module.
+
+Optional:
+* pluginImportPath - import path override, default `$module/plugin`
+* replace - go module replacement defined by `module,version`.
+
+Example of `.kn.yaml`
+```yaml
+plugins:
+  - name: kn-plugin-source-kafka
+    module: knative.dev/kn-plugin-source-kafka
+    pluginImportPath: knative.dev/kn-plugin-source-kafka/plugin
+    version: v0.19.0
+    replace:
+      - module: foo.bar
+        version: v0.0.1
+```
+
+
+Execute command
+```bash
+knb plugin distro
+```
+
+
+Build `kn`
+```bash
+./hack/build.sh
+```
+
+##### Enable plugin inline feature
+
+The `knb` can be used to generate required go files to inline any `kn` plugin.
+
+```bash
+knb plugin init --name kn-source-kafka --cmd source,kafka --description "Some plugin"
```

Comment:
Why not link to the generated docs for `knb` so that the options are in the natural expected fashion. For instance `—cmd` is an option but not clear what the possible values are...

Reply:
It was meant to represent one line example. The cmd docs are added below, but I can add the same generation of cmd descriptions to `./docs` dir of course.